koku/masu/database/ocp_report_db_accessor.py

```diff
@@ -1013,14 +1013,21 @@ def populate_openshift_cluster_information_tables(self, provider, cluster_id, cl
 
     def populate_cluster_table(self, provider, cluster_id, cluster_alias):
         """Get or create an entry in the OCP cluster table."""
+
         with schema_context(self.schema):
-            cluster, created = OCPCluster.objects.get_or_create(
-                cluster_id=cluster_id, cluster_alias=cluster_alias, provider=provider
-            )
+            clusters = OCPCluster.objects.filter(cluster_id=cluster_id, cluster_alias=cluster_alias, provider=provider)
+            cluster = clusters.first()
 
-        if created:
-            msg = f"Add entry in reporting_ocp_clusters for {cluster_id}/{cluster_alias}"
-            LOG.info(msg)
+            if cluster is None:
+                cluster = OCPCluster.objects.create(
+                    cluster_id=cluster_id, cluster_alias=cluster_alias, provider=provider
+                )
+                LOG.info(f"Add entry in reporting_ocp_clusters for {cluster_id}/{cluster_alias}")
+            elif clusters.count() > 1:
+                # If multiple clusters are found, delete all except the first one
+                OCPCluster.objects.filter(
+                    cluster_id=cluster_id, cluster_alias=cluster_alias, provider=provider
+                ).exclude(pk=cluster.pk).delete()
```

Comment:
I am not really sure we need to make any of these changes, especially this delete step. If we had a uniqueness constraint on this table, then we wouldn't need to worry about duplicate entries. If we added to `OCPCluster`:
```
    class Meta:
        """Meta for OCPCluster."""
        
        db_table = "reporting_ocp_clusters"
        unique_together = ("cluster_id", "cluster_alias", "provider")
```
this unique_together should prevent duplicate entries going forward.